crates/tests-e2e/tests/zzz_stop_orders.rs

```diff
@@ -0,0 +1,49 @@
+// XXX: This file starts with `zzz` as the tests are run in alphabetical order.
+// This test kills and starts the maker process again, and we do not want to interfere with other
+// tests.
+
+use native::api::ContractSymbol;
+use orderbook_commons::Prices;
+use tests_e2e::process::is_maker_running;
+use tests_e2e::process::kill_process;
+use tests_e2e::setup::TestSetup;
+use tests_e2e::wait_until;
+
+fn has_any_price(prices: &Option<Prices>) -> bool {
+    // Hardcoded to BTCUSD
+    if let Some(prices) = prices {
+        assert!(prices.len() == 1, "only BTCUSD is supported for now");
+        let btcusd = prices
+            .get(&ContractSymbol::BtcUsd)
+            .expect("BTCUSD price to be present");
+        return btcusd.bid.is_some() || btcusd.ask.is_some();
+    }
+    false
+}
+
+#[tokio::test]
+#[ignore = "need to be run with 'just e2e' command"]
+async fn no_orders_when_counterparty_is_offline() {
```

Comment:
The test name says that you should not expect an order if the counterparty is offline. This makes sense of course, but I guess what you are testing here is that outdated orders (older than 1m) are not beeing forwarded anymore? 

I don't think there is any use in testing if a terminated binary is not sending orders anymore. 

If you want to test expired orders, you can do a simpler test: 

1. Only start the coordinator and the app
2. Manually post a `NewOrder` to the coordinator using a reqwest client. Here you can set the expiry [to in Xsec](https://github.com/get10101/10101/blob/1eb9efbf3987175a0733fc78f9f21204a9149ca0/crates/orderbook-commons/src/lib.rs#L72).
3. Assert that no new order arrives past this few seconds. 

But even that test might not be that useful...

Reply:
I wanted to test that we detect lack of orders when the counterparty goes offline. As it turns out, we didn't. So I wrote the code that prunes outdated orders in the app so the feature would be working. 

I like your idea with 'faking' maker; we could have created one with different expiry.
I don't think that the test would be *simpler*, because we would need to kill the maker anyway, and start using a "fake" one, but it certainly would be faster!

I don't mind stripping out the test if you see no value in it.

Reply:
> I don't mind stripping out the test if you see no value in it.

Please treat me as an equal engineer, not as a your superior :)

Thinking about this more, can we write unit tests for this at the places where it matters? e.g. 
- [here](https://github.com/get10101/10101/blob/a1065b46aa6a0107bcdad9c82a6719072ced7ddd/mobile/native/src/orderbook.rs#L63-L80) we write a test that no expired order is being forwarded
- in the frontend we check if the order is still up2date when reading it.

Reply:
Perhaps let's rephrase it:
> I don't mind stripping out the test if you see no value in it.

What I meant was that I'm OK with stripping out the test if it causes contention for *any* of the reviewers, as it's not necessary for addressing the issue. 

I wrote the test because things were *supposed to* work, and didn't (normal part of engineering process). IMHO it would be great if we keep the test as it's proven to be useful.

I believe integration tests like this are needed; we need to know how we behave when the other party is not available and have that tested.

The criticism for waiting 70s is very much valid, and there's an open ticket that would address it - the maker doesn't have HTTP API to control orders for now, so it was not easy to address in this PR.

🤔Could I have started with an unit test instead? I don't think so - unit tests were there, it's just this case was *missed* during the implementation phase.

If you would like to add additional unit tests, or schedule them for the iteration, I think it's a good idea (the more unit tests the better!).

Reply:
@bonomat I have extracted the test into a separate PR, so it's not blocking the functionality:  #937